app/controllers/api/v2/registration_tokens_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V2
+    class RegistrationTokensController < V2::BaseController
+      include Foreman::Controller::UsersMixin
+
+      include Foreman::Controller::Parameters::User
+      include Foreman::Controller::AutoCompleteSearch
+      before_action :authenticate, :only => [:invalidate_jwt_tokens, :invalidate_jwt]
+
+      def resource_class
+        User
+      end
+
+      def find_resource(permission = :view_users)
+        editing_self? ? User.find(User.current.id) : User.authorized(permission).except_hidden.find(params[:id])
+      end
+
+      def action_permission
+        case params[:action]
+        when 'invalidate_jwt_tokens', 'invalidate_jwt'
+          'edit'
+        else
+          super
+        end
+      end
+
+      api :DELETE, '/users/:id/registration_tokens', N_("Invalidate all registration tokens for a specific user.")
+      description <<-DOC
+      The user you specify will no longer be able to register hosts by using their JWTs.
+      DOC
+      param :id, String, :desc => N_("ID of the user"), :required => true
+
+      def invalidate_jwt
+        @user = find_resource(:edit_users)
+        unless @user
+          raise ::Foreman::Exception.new(N_("No record found for %s"), params[:id])
+        end
+        @user.jwt_secret&.destroy
+        process_success _("Successfully invalidated registration tokens for %s.\n" % @user.login)
+      end
+
+      api :DELETE, "/registration_tokens", N_("Invalidate all registration tokens for multiple users.")
+      param :search, String, :required => true
```

Comment:
@Lennonka , could you please provide me with a description for the search parameters that I could use.

Reply:
`URL-encoded search query that selects users for which registration tokens will be invalidated. Search query example: id ^ (2, 4, 6)`